File: src/main/java/com/example/calendarexperiments/JsonConfig.java
package com.example.calendarexperiments;

import com.example.calendarexperiments.jackson.BitSetModule;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;

@Configuration
public class JsonConfig extends MappingJackson2HttpMessageConverter {
    public JsonConfig(ObjectMapper objectMapper) {
        super(objectMapper.registerModule(new BitSetModule()));
    }
}

File: src/main/kotlin/com/example/calendarexperiments/jackson/JsonConfig.kt
package com.example.calendarexperiments.jackson

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.context.annotation.Configuration
import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter

@Configuration
class JsonConfig(objectMapper: ObjectMapper) :
    MappingJackson2HttpMessageConverter(objectMapper.registerModule(BitSetModule()))